libreyolo/models/rfdetr/transformer.py

```
@@ -577,12 +802,37 @@ def forward(
         intermediate = []
         hs_refpoints_unsigmoid = [refpoints_unsigmoid]
 
+        # --- GroupPose keypoint additions (ported from RF-DETR v1.8.0). ---
+        keypoint_tgt = None
+        kp_query_pos = None
+        intermediate_keypoints = []
+
+        if self.enable_keypoint_processing:
+            assert self.lite_refpoint_refine, "Keypoint processing requires lite_refpoint_refine"
+            if tgt_keypoints is None:
+                raise ValueError("Keypoint processing is enabled but tgt_keypoints was not provided")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)